crates/sui-pg-db/src/db.rs

```diff
@@ -0,0 +1,547 @@
+            END LOOP;
+        END $$;";
+        diesel::sql_query(drop_all_procedures)
+            .execute(&mut conn)
+            .await?;
+        info!("Dropped all procedures.");
+
+        let drop_all_functions = "
+        DO $$ DECLARE
+            r RECORD;
+        BEGIN
+            FOR r IN (SELECT proname, oidvectortypes(proargtypes) as argtypes
+                      FROM pg_proc INNER JOIN pg_namespace ON (pg_proc.pronamespace = pg_namespace.oid)
+                      WHERE pg_namespace.nspname = 'public' AND prokind = 'f')
+            LOOP
+                EXECUTE 'DROP FUNCTION IF EXISTS ' || quote_ident(r.proname) || '(' || r.argtypes || ') CASCADE';
+            END LOOP;
+        END $$;";
+        diesel::sql_query(drop_all_functions)
+            .execute(&mut conn)
+            .await?;
+        info!("Database cleared.");
+        Ok(())
+    }
+
+    /// Run migrations on the database. Use Diesel's `embed_migrations!` macro to generate the
+    /// `migrations` parameter for your indexer.
+    pub async fn run_migrations<S: MigrationSource<Pg> + Send + Sync + 'static>(
+        &self,
+        migrations: S,
+    ) -> anyhow::Result<Vec<MigrationVersion<'static>>> {
+        use diesel_migrations::MigrationHarness;
+
+        info!("Running migrations ...");
+        let conn = self.pool.dedicated_connection().await?;
+        let mut wrapper: AsyncConnectionWrapper<AsyncPgConnection> =
+            diesel_async::async_connection_wrapper::AsyncConnectionWrapper::from(conn);
+
+        let finished_migrations = tokio::task::spawn_blocking(move || {
+            wrapper
+                .run_pending_migrations(migrations)
+                .map(|versions| versions.iter().map(MigrationVersion::as_owned).collect())
+        })
+        .await?
+        .map_err(|e| anyhow!("Failed to run migrations: {:?}", e))?;
+
+        info!("Migrations complete.");
+        Ok(finished_migrations)
+    }
+}
+
+impl Default for DbArgs {
+    fn default() -> Self {
+        Self {
+            db_connection_pool_size: 100,
+            connection_timeout_ms: 60_000,
+        }
+    }
+}
+
+impl<'a> Deref for DbConnection<'a> {
+    type Target = Connection<'a>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for DbConnection<'_> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[async_trait]
+impl StoreConnection for DbConnection<'_> {
+    async fn committer_watermark(
+        &mut self,
+        pipeline: &'static str,
+    ) -> anyhow::Result<Option<CommitterWatermark>> {
+        let watermark: Option<StoredWatermark> = watermarks::table
+            .select(StoredWatermark::as_select())
+            .filter(watermarks::pipeline.eq(pipeline))
+            .first(self)
+            .await
+            .optional()
+            .map_err(anyhow::Error::from)?;
```

Comment:
You shouldn't need to explicitly call `anyhow::Error::from` here and elsewhere in this file -- `?` does this automatically. What happens when you get rid of this?